share/zephyr-package/cmake/zephyr_package_search.cmake

```diff
@@ -0,0 +1,78 @@
+# The purpose of this file is to provide search mechanism for locating Zephyr in-work-tree package
```

Comment:
Please consider contributing this upstream to CMake.

Reply:
That's a great idea.
Although, it would only help users that are using Zephyr repository apps or Zephyr workspace apps.

A Zephyr freestanding app would still need to have a Zephyr registered, in order to know which Zephyr to use.

Let's get this PR in, and I will continue to explore possibilities for your suggestion :smile: